src/prefect/utilities/collections.py

```diff
@@ -29,6 +29,8 @@
 
 import pydantic
 
+from prefect.settings import PREFECT_PARAMETERS_TRAVERSE_PYDANTIC_MODELS
```

Comment:
This seems like it ought to live in the engine and `visit_collection` should take a set of types to exclude traversal of. However, it's simpler to put it here and I'm not sure how much the purity matters.